### PR TITLE
Aks pipeline changes

### PR DESCRIPTION
**Pull request by mrganeshraja**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
-    NODE_ENV=$NODE_ENV
+ENV NODE_ENV=$NODE_ENV
+ENV NODE_PATH=/opt
 
-COPY . /opt/app
-RUN yarn && yarn setup && yarn cache clean
+COPY . /opt
+
+RUN apk update && \
+    apk add --no-cache bash make g++ git python2
```
Comment (timwebster9) [inline Dockerfile]: why is this line required?  What are these OS dependencies for?

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,21 @@
...
+
+RUN yarn install --production=false && yarn setup
+
+RUN rm -rf node_modules \
```
Comment (timwebster9) [inline Dockerfile]: Why is `yarn install` run a second time here?

File: Dockerfile [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
+ENV NODE_PATH=/opt
 
-COPY . /opt/app
-RUN yarn && yarn setup && yarn cache clean
+COPY . /opt
+
+RUN apk update && \
+    apk add --no-cache bash make g++ git python2
+
+RUN yarn install --production=false && yarn setup
```
Comment (timwebster9) [inline Dockerfile]: why is `--production` set to false here (and `NODE_ENV` is defaulting to `production`?

File: Dockerfile [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
+
+RUN rm -rf node_modules \
+    && yarn install \
```
Comment (timwebster9) [inline Dockerfile]: Why is `yarn install` run a second time here?

File: Dockerfile [hunk 5]
```diff
@@ -1,14 +1,21 @@
-FROM divorce/yarn:8
+FROM node:8.9.4-alpine
```
Comment (timwebster9) [inline Dockerfile]: Any reason not to use a newer version of Node?  E.g. 10+?

File: Dockerfile (hunk 4 above)
Comment (timja) [reply Dockerfile]: he will need some dependencies in devDependencies such as gulp to run his setup command

similar to the cmc one here:
https://github.com/hmcts/cmc-citizen-frontend/blob/master/Dockerfile

Comment (timwebster9) [reply Dockerfile]: OK but I still don't get why it needs to run twice...

**Comment by hosainnet** [comment]
AKS already implemented
